src/messaging/ExchangeMgr.h

```diff
@@ -48,6 +48,8 @@ static constexpr int16_t kAnyMessageType = -1;
  */
 class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate
 {
+    friend class MessageCounterSyncMgr;
```

Comment:
why is a friend class necessary? Can it be avoided?

Friend classes often indicate tight coupling.